twistlock/datadog_checks/twistlock/twistlock.py

```diff
@@ -0,0 +1,322 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from collections import defaultdict
+import time
+from datetime import datetime, timedelta
+
+from six import iteritems
+
+import requests
+
+from datadog_checks.base import AgentCheck
+
+from .config import Config
+
+REGISTRY_SCAN_DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+SCAN_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+LICENSE_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+DOCKERIO_PREFIX = "docker.io/"
+
+# min_severity tag allows to query number of vulns for a severity target easily
+SEVERITY_TAGS = {
+    "low": ["severity:low", "min_severity:low"],
+    "medium": ["severity:medium", "min_severity:low", "min_severity:medium"],
+    "high": ["severity:high", "min_severity:low", "min_severity:medium", "min_severity:high"],
+    "critical": ["severity:critical", "min_severity:low", "min_severity:medium",
+                 "min_severity:high", "min_severity:critical"]
+}
+
+
+class TwistlockCheck(AgentCheck):
+    NAMESPACE = 'twistlock'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+        msg = 'This check has not run before, '
+        msg += 'it will send all the new vulnerabilities that happened in the past day'
+        self.log.debug(msg)
+        self.last_run = datetime.now() - timedelta(days=1)
```

Comment:
Why this logic, why 1 day?

Reply:
arbitrary choice, seemed like a good time frame. What do you think is better?

Reply:
If there are no rules, is that something we should make configurable with defaults?

Reply:
I say default to nothing so an Agent restart doesn't submit duplicate metrics. That is bad.

Reply:
It doesn't send any metrics, only events

Reply:
Same critique, events don't get deduped

Reply:
the worry is that if it frequently restarts, as container agents do, it will never send a lot of events

Reply:
But i'll remove it